python/pyspark/sql/dataframe.py

```diff
@@ -375,7 +375,12 @@ def _repr_html_(self):
         by 'spark.sql.repl.eagerEval.enabled', this only called by REPL you are
         using support eager evaluation with HTML.
         """
-        import cgi
+        try:
```

Comment:
I think we have some version-specific imports up at the top of this file, e.g.

https://github.com/apache/spark/blob/560aa43736dcf85f7e49391ce8fe0a8da8d30f69/python/pyspark/sql/dataframe.py#L21

so maybe we could group those there?

On the other hand, the current behavior is to only load `cgi` if `_repr_html_` is invoked and this PR's changes preserve that behavior, whereas moving it to the top would cause it to be unconditionally imported. Unconditional import is fine if this import is generally side-effect-free.

Don't feel strongly, just wanted to note that.

Regardless of whether we move, should we do an explicit `sys.version` check here instead of the `try except`?

Reply:
+1 for @JoshRosen 's comment.

Reply:
Thanks @JoshRosen @dongjoon-hyun for the advice! I will follow it.

Reply:
I believe either way is fine per https://docs.python.org/3/howto/pyporting.html#use-feature-detection-instead-of-version-detection but yea, let's use pre-defined condition then.